tests/edit_parameters.cc

```diff
@@ -0,0 +1,66 @@
+#include <deal.II/base/parameter_handler.h>
+#include <aspect/global.h>
+#include <aspect/simulator_signals.h>
+
+namespace aspect
+{
+  using namespace dealii;
+
+  // Global variables (to be set by parameters)
+  int switch_step;
+  bool switched;
+
+  /**
+   * Declare additional parameters.
+   */
+  void declare_parameters(const unsigned int,
+                          ParameterHandler &prm)
+  {
+    prm.declare_entry("Switch step", "0",
+                       Patterns::Integer(0),
+                       "Switch the bottom boundary condition from fixed temperature to no-flux"
+                       "at the timestep given.");
+  }
+
+  template <int dim>
+  void parse_parameters(const Parameters<dim>,
+                        ParameterHandler &prm)
+  {
+    switch_step = prm.get_integer("Switch step");
+    switched = false;
+  }
+
+  template <int dim>
+  void change_boundary_condition (const SimulatorAccess<dim> &simulator_access,
+                                  Parameters<dim> &parameters)
+  {
+    types::boundary_id bottom = 2; //bottom boundary indicator for 2D box
+
+    if ( simulator_access.get_timestep_number() >= switch_step && !switched )
+      {
```

Comment:
ah, I see -- you want to have the signal called at the beginning of each timestep. Isn't there a `start_timestep()` or `begin_timestep()` function or similar from which this signal could be called?

I think it would be worthwhile documenting in the description of the signal where and when the signal is triggered.